product-service/functions/catalogBatchProcess.js

```diff
@@ -0,0 +1,79 @@
+const AWS = require("aws-sdk");
+const uuid = require("uuid");
+const dynamoDB = new AWS.DynamoDB.DocumentClient();
+const sns = new AWS.SNS();
+
+const productTable = process.env.PRODUCT_TABLE;
+const stockTable = process.env.STOCK_TABLE;
+const snsTopicArn = process.env.SNS_TOPIC_ARN;
+
+const createProduct = async (product) => {
+  const params = {
+    TableName: productTable,
+    Item: product,
+  };
+
+  try {
+    await dynamoDB.put(params).promise();
+    console.log(`Product created: ${JSON.stringify(product)}`);
+
+    await sns.publish({
```

Comment:
It sees this has to be done only once, when all the products were created, but even if its done for each one of these it seems well implemented, so its excellent. Just wanted to leave a note about this.